Iss hipcms 501 tag relations

janis-kra opened the pull request:
Do we also need a PUT method that allows changing the name of a tag or is that out of the scope of this task?

maltehol commented:
Not sure if you get the concept correctly.
Relations will be set between instantiated Tags in a text. 
That means, for each Tag, there will be multiple instantiated Tags with a separate id.

To get not confused I distinguish on Client side between `TagModel` and `Tag`. `Tag Model` is what already is on Server site. A `Tag` has a `TagModel`, an `Id` and a `RelationTo`.

janis-kra commented:
@maltehol I definitely did not get the concept correctly then - can I read up on this somewhere in our wiki?

@sstamm thanks for the input - about the bidirectional relation thing: I see two alternatives, storing every relation twice (as I implemented) or store it once (as firstTag=A and secondTag=B or sth like that) and then add a second check in order to find this relation even when the client asks for the relation as firstTag=B and secondTag=A (which is the same in terms of our business logic, but not for the database). As a side-benefit, the approach I implemented supports uni-directional dependencies just in case they are demanded some time in the future.

janis-kra commented on src/Api/Controllers/AnnotationController.cs:
```diff
@@ -145,22 +145,40 @@ public IActionResult PostChildTag(int parentId, int childId)
         [ProducesResponseType(typeof(void), 403)]
         [ProducesResponseType(typeof(void), 400)]
-        public IActionResult PostTagRelation(int firstId, int secondId, string name)
+        public IActionResult PostTagRelation(int firstId, int secondId, string name = "")
```
@sstamm any idea why the name parameter is always the default value when I test this via the swagger ui?

janis-kra commented:
@maltehol feel free to take a look at the changed model -- hopefully I got it right this time

janis-kra commented on src/Api/Controllers/AnnotationController.cs:
```diff
@@ -145,6 +145,38 @@ public IActionResult Post(int parentId, int childId)
+        [ProducesResponseType(typeof(void), 200)]
+        [ProducesResponseType(typeof(void), 403)]
+        [ProducesResponseType(typeof(void), 400)]
+        public IActionResult PostTagRelation(int firstId, int secondId, string name = "")
```
When testing this with postman, there are no issues (i.e. the `name` is correctly set). So there seems to be an issue with swagger misinterpreting something.

Adding [FromBody] to the parameter declaration does not help, this causes the parameter to disappear from the swagger doc. Also, adding [FromRoute] to the firstId and secondId parameters - which I suppose would be the correct way of declaring the webservice endpoint - makes the whole call disappear from the swagger doc. Thoughts?

janis-kra commented:
I'll just close this and reopen as soon as I'm finished
